tasks/Functions-billing/fund.js

```diff
@@ -8,60 +11,32 @@ task("functions-sub-fund", "Funds a billing subscription for Functions consumer
       throw Error("This command cannot be used on a local hardhat chain.  Specify a valid network.")
     }
 
-    const subscriptionId = taskArgs.subid
-    const linkAmount = taskArgs.amount
-
-    const RegistryFactory = await ethers.getContractFactory(
-      "contracts/dev/functions/FunctionsBillingRegistry.sol:FunctionsBillingRegistry"
-    )
-    const registry = await RegistryFactory.attach(networks[network.name]["functionsBillingRegistryProxy"])
-
-    // Check that the subscription is valid
-    let preSubInfo
-    try {
-      preSubInfo = await registry.getSubscription(subscriptionId)
-    } catch (error) {
-      if (error.errorName === "InvalidSubscription") {
-        throw Error(`Subscription ID "${subscriptionId}" is invalid or does not exist`)
-      }
-      throw error
-    }
+    const signer = await ethers.getSigner()
+    const linkTokenAddress = networks[network.name]["linkToken"]
+    const functionsRouterAddress = networks[network.name]["functionsRouter"]
+    const txOptions = { confirmations: networks[network.name].confirmations }
 
-    // Convert LINK to Juels
-    const juelsAmount = ethers.utils.parseUnits(linkAmount)
-    console.log(`Funding subscription ${subscriptionId} with ${ethers.utils.formatEther(juelsAmount)} LINK`)
+    const subId = taskArgs.subid
+    const linkAmount = taskArgs.amount
+    const juelsAmount = ethers.utils.parseUnits(linkAmount, 18)
 
-    const LinkTokenFactory = await ethers.getContractFactory("LinkToken")
-    const linkToken = await LinkTokenFactory.attach(networks[network.name].linkToken)
+    const sm = new SubscriptionManager({ signer, linkTokenAddress, functionsRouterAddress })
+    await sm.initialize()
 
-    const accounts = await ethers.getSigners()
-    const signer = accounts[0]
+    await utils.prompt(
+      `\nPlease confirm that you wish to fund Subscription '${subId}' with '${linkAmount}' LINK from your wallet.'`
+    )
 
-    // Ensure sufficient balance
-    const balance = await linkToken.balanceOf(signer.address)
-    if (juelsAmount.gt(balance)) {
-      throw Error(
-        `Insufficient LINK balance. Trying to fund subscription with ${ethers.utils.formatEther(
-          juelsAmount
-        )} LINK, but wallet only has ${ethers.utils.formatEther(balance)}.`
-      )
-    }
+    console.log(`\nFunding subscription '${subId}' with '${linkAmount}' LINK...`)
 
```

Comment:
convert linkAmount -> juelsAmount

Reply:
For the subscription that has been done on line 21.

the logging can be done in LINK because the HH kit user put in the sub amount in LINK.